hw/formal/tools/jaspergold/parse-fpv-report.py

```diff
@@ -0,0 +1,217 @@
+        results["fpv_summary"] = fpv_summary
+
+    return results
+
+
+def get_cov_results(logpath):
+    '''Parse FPV coverage information from the log file'''
+    cov_results = OrderedDict()
+    try:
+        with Path(logpath).open() as f:
+            full_file = f.read()
+            coverage_patterns = [("stimuli", r"^\|\w+.*\|(\d*.\d\d)%.*\|\d*.\d\d%.*\|\d*.\d\d%.*"),
+                                 ("coi", r"^\|\w+.*\|\d*.\d\d%.*\|(\d*.\d\d)%.*\|\d*.\d\d%.*"),
+                                 ("proof", r"^\|\w+.*\|\d*.\d\d%.*\|\d*.\d\d%.*\|(\d*.\d\d)%.*")]
+            cov_results.update(extract_messages(full_file, coverage_patterns))
+            for key, item in cov_results.items():
+                if len(item) == 1:
+                    cov_results[key] = item[0] + " %"
+                else:
+                    cov_results[key] = "N/A"
+                    log.error("Parse %s coverage error. Expect one matching value, get %s",
+                              key, item)
+
+    except IOError as err:
+        log.error("[get_cov_results] IOError %s", err)
+        cov_results = None
+
+    return cov_results
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description=
+        '''This script parses the JasperGold FPV log to extract below information.
+
+        "messages": {
+          "errors"      : []
+          "warnings"    : []
+          "cex"         : ["property1", "property2"...],
+          "undetermined": [],
+          "unreachable" : [],
+        },
+        "fpv_summary": {
+          "errors"      : 0
+          "warnings"    : 2
+          "proven"      : 20,
+          "cex"         : 5,
+          "covered"     : 18,
+          "undetermined": 7,
+          "unreachable" : 2,
+          "pass_rate"   : "90 %",
+          "cover_rate"  : "90 %"
+        },
+        If coverage is enabled, this script will also parse the coverage result:
+        "fpv_coverage": {
+          stimuli: "90 %",
+          coi    : "90 %",
+          proof  : "80 %"
+        }
+        The script returns nonzero status if any errors or property failures including
+        "cex, undetermined, unreachable" are presented.
+        ''')
+    parser.add_argument('--logpath',
+                        type=str,
+                        default="fpv.log",
+                        help=('FPV log file path. Defaults to `fpv.log` '
+                              'under the current script directory.'))
+
+    parser.add_argument('--reppath',
+                        type=str,
+                        default="results.hjson",
+                        help=('Parsed output hjson file path. Defaults to '
+                              '`results.hjson` under the current script directory.'))
+
+    parser.add_argument('--cov',
+                        type=int,
+                        default=0,
+                        help=('Enable parsing coverage data. '
+                              'Default coverage parsing is disabled.'))
```

Comment:
I think you probably want "By default, coverage parsing is disabled" (otherwise it sounds like we don't do "default coverage parsing").